### PR TITLE
Update jQuery

### PR DESCRIPTION
**Pull request by ryuyu**
Taking the update to jQuery 3.x

Mitigates an XSS attack.

Removed reference to removed apis.
Used https://github.com/jquery/jquery-migrate/blob/master/warnings.md#jqmigrate-jqueryfndelegate-is-deprecated for warning and https://jquery.com/upgrade-guide/3.0/ for upgrade assistance.

Did not remove use of deprecated apis.

### CODE REVIEW

File: src/NuGetGallery/App_Start/AppActivator.cs [hunk 1]
```diff
@@ -141,9 +141,10 @@ private static void BundlingPostStart()
             BundleTable.Bundles.Add(newStyleBundle);
 
             var scriptBundle = new ScriptBundle("~/Scripts/gallery/site.min.js")
-                .Include("~/Scripts/gallery/jquery-1.12.4.js")
+                .Include("~/Scripts/gallery/jquery-3.4.1.js")
                 .Include("~/Scripts/gallery/jquery.validate-1.16.0.js")
                 .Include("~/Scripts/gallery/jquery.validate.unobtrusive-3.2.6.js")
+                .Include("~/Scripts/gallery/jquery-migrate-3.0.js")
```
Comment (joelverhagen) [inline src/NuGetGallery/App_Start/AppActivator.cs]: I don't think we should ship this, right?

Comment (joelverhagen) [reply src/NuGetGallery/App_Start/AppActivator.cs]: My understanding was that this is just a diagnostic tool.

Comment (ryuyu) [reply src/NuGetGallery/App_Start/AppActivator.cs]: Yes should be removed. Whoops!

File: src/NuGetGallery/NuGetGallery.csproj [hunk 2]
```diff
@@ -1765,7 +1765,40 @@
     <Content Include="Areas\Admin\Views\Features\_EditFlight.cshtml" />
     <Content Include="App_Data\Files\Content\AB-Test-Configuration.json" />
     <None Include="Properties\PublishProfiles\nuget-staging-frontend.pubxml" />
-    <Content Include="Scripts\gallery\**\*" />
+    <Content Include="Scripts\gallery\async-file-upload.js" />
```
Comment (joelverhagen) [inline src/NuGetGallery/NuGetGallery.csproj]: Should change can be undone.

File: src/NuGetGallery/NuGetGallery.csproj [hunk 3]
```diff
@@ -1765,7 +1765,40 @@
...
+    <Content Include="Scripts\gallery\autocomplete.js" />
+    <Content Include="Scripts\gallery\bootstrap.js" />
+    <Content Include="Scripts\gallery\certificates.js" />
+    <Content Include="Scripts\gallery\clamp.js" />
+    <Content Include="Scripts\gallery\common-multi-select-dropdown.js" />
+    <Content Include="Scripts\gallery\common.js" />
+    <Content Include="Scripts\gallery\jquery-1.12.4.js" />
```
Comment (joelverhagen) [inline src/NuGetGallery/NuGetGallery.csproj]: We can remove 1.12.4 right?

Comment (ryuyu) [reply src/NuGetGallery/NuGetGallery.csproj]: Yes i removed it.

Comment (cristinamanum) [reply src/NuGetGallery/NuGetGallery.csproj]: Was testing executed on a private deployment? I think that it would be good.

Comment (ryuyu) [reply src/NuGetGallery/NuGetGallery.csproj]: Yes testing was executed on a private deployment of gallery.

**Pull request merged**